add EXT-X-KEY tag support to playlist parser

Lysander66 opened the pull request:
Media Segments MAY be encrypted.

Add support for EXT-X-KEY tag in m3u8 playlist parser.

Ref: https://datatracker.ietf.org/doc/html/rfc8216#section-4.3.2.4

aler9 commented:
Hello, thanks for this patch, just fix lint errors and add a couple of tests and it will be merged.

Lysander66 commented:
> Hello, thanks for this patch, just fix lint errors and add a couple of tests and it will be merged.

Hi, I've fixed the lint errors and added the test cases for EXT-X-KEY tag with different scenarios.
